src/gallery.js

```diff
@@ -1,19 +1,101 @@
+import { msg } from './i18n'
+
 let gallery = [],
 	current = 0
 
-const hideFullscreenGallery = ( galleryContainer ) => {
+const getFullScreenGallery = () => {
+		return `
+			<div class="wp-gallery-popup">
+				<div class="wp-gallery-popup-top">
+					<div class="wp-gallery-popup-button close"></div>
+				</div>
+				<div class="wp-gallery-popup-main">
+					<div class="wp-gallery-popup-button previous"></div>
+					<div class="wp-gallery-popup-image">
+						<img src="">
+						<div class="wp-gallery-popup-loading">
+							<div class="wp-gallery-popup-loading-icons">
+								<div class="wp-gallery-popup-loading-spinner">
+									<div class="wp-gallery-popup-loading-spinner-animation"></div>
+								</div>
+								<div class="wp-gallery-popup-loading-error"></div>
+							</div>
+							<div class="wp-gallery-popup-loading-text"></div>
+							<div class="wp-gallery-popup-loading-error-refresh"></div>
+						</div>
+					</div>
+					<div class="wp-gallery-popup-button next"></div>
+				</div>
+				<div class="wp-gallery-popup-bottom">
+					<div class="wp-gallery-popup-caption"></div>
+				</div>
+			</div>
+		`.trim()
+	},
+
+	toggleLoading = ( loading, image, lang ) => {
+		const text = loading.querySelector( '.wp-gallery-popup-loading-text' ),
+			error = loading.querySelector( '.wp-gallery-popup-loading-error' ),
+			refresh = loading.querySelector( '.wp-gallery-popup-loading-error-refresh' ),
+			spinner = loading.querySelector( '.wp-gallery-popup-loading-spinner' ),
+
+			timeoutId = setTimeout( () => {
+				text.innerHTML = msg( lang, 'gallery-loading-still' )
+				text.style.visibility = 'visible'
+			}, 5000 ),
+
+			onLoad = () => {
+				clearTimeout( timeoutId )
+				loading.style.visibility = 'hidden'
+				text.style.visibility = 'hidden'
+				image.style.visibility = 'visible'
+				image.removeEventListener( 'load', onLoad )
+			},
+
+			onRefresh = () => {
+				toggleLoading( loading, image )
+				image.src = gallery[ current ].src
+				refresh.removeEventListener( 'click', onRefresh )
+			},
+
+			onError = () => {
+				clearTimeout( timeoutId )
+				error.style.visibility = 'visible'
+				spinner.style.visibility = 'hidden'
+				text.innerHTML = msg( lang, 'gallery-loading-error' )
```

Comment:
Please use `innerText` as much as possible so it is not vulnerable to HTML injection.